src/commands/command.ts

```diff
@@ -1,102 +1,96 @@
+    channelLabel: (channel: Channel) => string,
+    channelHelp: (channel: Channel, prefix: string, role?: UserRole) => string,
+    channelTrigger: (channel: Channel) => RegExp,
+    action: (message: Message, match: RegExpMatchArray) => Promise<void>,
     role?: UserRole,
-    hasPrefix?: boolean,
   ) {
-    this.label = label;
+    this.name = name;
     this.description = description;
-    this.triggerLabel = triggerLabel;
-    this.trigger = trigger;
-    this.callback = callback;
+    this.channelLabel = channelLabel;
+    this.channelHelp = channelHelp;
+    this.channelTrigger = channelTrigger;
+    this.action = action;
     this.role = role != null ? role : UserRole.USER;
-    this.hasPrefix = hasPrefix != null ? hasPrefix : true;
   }
   /** Tries to execute the command on the given channel.
    *
-   * @param bot - The BotClient to execute the command on.
-   * @param channel - The channel to execute the command on.
-   * @param user - The user trying to execute the command.
+   * @param message - The message triggering the command.
    * @param match - The RegExp match of the command trigger.
    */
-  public async execute(
-    bot: BotClient,
-    message: Message,
-    match: RegExpMatchArray,
-  ): Promise<boolean> {
+  public async execute(message: Message, match: RegExpMatchArray): Promise<boolean> {
     // Check if the user has the required role to execute the command.
+    const bot = message.channel.bot;
+
     if (await message.user.hasRole(message.channel, this.role)) {
-      await this.callback(bot, message, match);
+      await this.action(message, match);
       const time = Date.now() - message.timestamp.valueOf();
-      bot.logger.debug(`Command '${this.label}' executed in ${time} ms.`);
+      if (!(this instanceof CommandGroup)) {
+        bot.logger.debug(`Command '${this.name}' executed in ${time} ms.`);
+      }
       return true;
     }
-    bot.logger.debug(`Command: ${this.label}: ${this.role} role required.`);
+
+    bot.logger.debug(`Command: ${this.name}: ${this.role} role required.`);
     bot.sendMessage(
       message.channel,
       `You need the role '${this.role}' on this server to execute this command!`,
     );
+
     return false;
   }
 
   /** Gets the RegExp used to trigger the command
    *
-   * @param client - The BotChannel to trigger the command on.
+   * @param channel - The channel to get the regex for.
    */
-  public async getRegExp(channel: Channel) {
-    const bot = channel.client;
-    const userTag = EscapeRegex(await bot.getUserTag());
-    const channelPrefix = EscapeRegex(channel.getPrefix());
-    const prefix = this.hasPrefix
-      ? `^\\s*((${userTag})|((${channelPrefix})(\\s*${userTag})?)|((${bot.prefix})\\s*(${userTag})))\\s*`
-      : '';
-
-    const regexString = prefix + this.trigger;
-    return new RegExp(regexString);
+  public getRegExp(channel: Channel): RegExp {
+    const regexString = regexToString(this.channelTrigger(channel));
```

Comment:
There is a simpler way to do that https://stackoverflow.com/questions/9213237/combining-regular-expressions-in-javascript

Reply:
Thanks, fixed.